priority_queue/types.mbt

```diff
@@ -0,0 +1,26 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+priv struct Cont[T] {
+  content : T
+  mut sibling : Node[T]
+  mut child : Node[T]
+}
+
+priv enum Node[T] {
+  Nil
+  Cons(Cont[T])
+}
+
+struct PriorityQueue[T] {
+  min : Bool
+  mut len : Int
```

Comment:
@Lampese for the API design, I feel `min` is not needed here, it slows down the algorithm a little bit and user can always create a new type with different comparison behavior

Reply:
Allright, but my initial consideration was that it might not be user-friendly to define Num for structures that already have a Compare relationship. But it doesn't really hurt to provide only one heap though, as many implementations do.

Reply:
Do I need to re-initiate a PR after this one is over to resolve this issue?